packages/utils/telemetry-utils/src/config.ts

```diff
@@ -54,6 +56,20 @@ export const inMemoryConfigProvider = (storage: Storage | undefined): IConfigPro
 				} catch {}
 				return undefined;
 			},
+			getRawConfigEntries(): [string, string][] {
+				const configEntries: Map<string, string> = new Map();
+				for (let i = 0; i < storage.length; i++) {
+					const key = storage.key(i);
```

Comment:
i don't think you can do this. it will enumerate all entries in session storage, we can't just log arbitray data from the browser

Reply:
i still think it's risky. i prefer log on first read.

Reply:
Ok, let me try that, it's going to be relatively noisy if we log on first read.

Reply:
We have approximately 43 feature flags. (based on a search of "Fluid.")

Reply:
overall, configs should be transient, we use them to ship dark and test, but then remove them. in the long term across all clients i expect there will be far fewer sessions with configs, than without.

Reply:
but for most sessions the values are not set, i would not log undefined values.

Reply:
I'd say a majority of them are called in constructors, so the few that are called on first read are likely to be sparse. I'll be sending somewhere of upwards of 30 telemetry events per new container load.

Reply:
i don't think you understand what i am proposing
``` diff
	private getCacheEntry(name: string): StronglyTypedValue | undefined {
		if (!this.configCache.has(name)) {
			for (const provider of this.orderedBaseProviders) {
				const parsed = stronglyTypedParse(provider?.getRawConfig(name));
				if (parsed !== undefined) {
					this.configCache.set(name, parsed);
+					this.logger?.send({
+						category: "generic",
+						eventName: "configRead",
+						configName: { tag: TelemetryDataTag.CodeArtifact, value: name },
+						configValue: {
+							tag: TelemetryDataTag.CodeArtifact,
+							value: JSON.stringify(parsed),
+						},
+					});
					return parsed;
				}
			}
			// configs are immutable, if the first lookup returned no results, all lookups should
			this.configCache.set(name, { raw: undefined });
		}
		return this.configCache.get(name);
	}
}

```

Reply:
it also has the benefit of being transparent to our partners, they just have to consume the version, and the logs will start flowing.